planner/core/plan.go

```diff
@@ -386,8 +386,13 @@ func (p *baseLogicalPlan) ExtractFD() *fd.FDSet {
 		return p.fdSet
 	}
 	fds := &fd.FDSet{HashCodeToUniqueID: make(map[string]int)}
+	// isolation between different logical selection blocks.
+	acrossBlock := false
```

Comment:
Would we need this check? If we ignore it, we would not add some wrong functional dependencies into the set?

Reply:
if not, we will inherit subquery's hasAgg flag and group by columns, and do the fd check in any projection of outer query.